configure.py

```diff
@@ -753,7 +753,7 @@ def find_ninja():
                         help='List all available build artifacts, that can be passed to --with')
 arg_parser.add_argument('--date-stamp', dest='date_stamp', type=str,
                         help='Set datestamp for SCYLLA-VERSION-GEN')
-arg_parser.add_argument('--use-cmake', action=argparse.BooleanOptionalAction, default=False, help='Whether to use CMake as the build system')
+arg_parser.add_argument('--use-cmake', action=argparse.BooleanOptionalAction, default=True, help='Whether to use CMake as the build system')
```

Comment:
We need proof that cmake passes the same compile options as the traditional system.

Reply:
i compared the compilation database generated from CMake and `configure.py` using the script at https://gist.github.com/tchaikov/eb2cdb08c013506346cd0c476f5b45da. the result can be found at https://gist.github.com/tchaikov/c7d5b79b7b8dfd9144ec3a27b79c4371

Reply:
> I looked at the diff and the only significant difference is this:
> 
> ```diff
> === wasmtime_bindings.cc:
> -{'W': set(), 'I': {'build/gen'}, 'f': set(), 'D': set()}
> +{'W': {'error=unused-result'}, 'I': {'build/Dev/seastar/gen/include', 'seastar/include'}, 'f': {'stack-clash-protection'}, 'D': {'SEASTAR_SCHEDULING_GROUPS_COUNT=16', 'SEASTAR_LOGGER_TYPE_STDOUT', 'SEASTAR_API_LEVEL=7', 'SEASTAR_SSTRING', 'SEASTAR_TYPE_ERASE_MORE', 'WITH_GZFILEOP', 'SEASTAR_LOGGER_COMPILE_TIME_FMT', 'FMT_SHARED'}}
> 
> === inc.cc:
> -{'W': set(), 'I': {'build/gen'}, 'f': set(), 'D': set()}
> +{'W': {'error=unused-result'}, 'I': {'build/Dev/seastar/gen/include', 'seastar/include'}, 'f': {'stack-clash-protection'}, 'D': {'SEASTAR_SCHEDULING_GROUPS_COUNT=16', 'SEASTAR_LOGGER_TYPE_STDOUT', 'SEASTAR_API_LEVEL=7', 'SEASTAR_SSTRING', 'SEASTAR_TYPE_ERASE_MORE', 'WITH_GZFILEOP', 'SEASTAR_LOGGER_COMPILE_TIME_FMT', 'FMT_SHARED'}}
> ```
> 
> Can you explain this?
> 

sure. these two files are both rust's C++ bindings generated using cxxbridge.

`build/gen`
---

in this very case, `configure.py` puts the rust module's C++ bindings into `build/<build_mode>/gen/rust`. while CMake puts them into `build/gen/rust`. as all modes share the same generated C++ bindings. BTW, we put the generated object files into per-build-mode subdirectories. and the compare script normalize the path used by `configure.py` by removing dirname, see https://gist.github.com/tchaikov/eb2cdb08c013506346cd0c476f5b45da .

seastar flags
---

rust binding does not use Seastar headers or link against Seastar libraries, so there is no need to build with Seastar cflags. this is also why i claimed 

> The variations mostly represent cleanup opportunities in the CMake configuration

in previous comment in this thread.


> BTW which is `-` and which is `+`?

i wanted to mimic the format of `diff` when comparing two building command databases. the two building command databases are generated using `configure.py` and CMake repectively, and the command is
```bash
../compare-compile-commands.py -ppppp {cmake-,}compile_commands.json
```

so the lines started with `-` is the command line options only found in the rules generated by CMake, and the lines started with `+` are those only found in the rules generated by `configure.py`. as you noticed, `configure.py` tends to pass more cflags when building object files even when they are not used at all.

> 
> This `'I': {'build/gen'},` is present everywhere, why is that?

actually, both of them have `build/gen`. but i am too lazy to normalize them. 

- the one from CMake looks like `-I/home/kefu/dev/scylladb/master -I/home/kefu/dev/scylladb/master/build/gen` 
- the one from `configure.py` looks like `-iquote. -iquote build/dev/gen`.

Reply:
Ok, it indeed looks like the generated build rules are identical where it counts. I'll queue this. :crossed_fingers:

Reply:
This is breaking gating, I assume because we need this change together with some change in `scylla-pkg`

Reply:
@scylladb/scylla-maint can you please dequeue this patch? it's breaking gating